File: cmk/checkers/_typedefs.py
#!/usr/bin/env python3
# Copyright (C) 2019 Checkmk GmbH - License: GNU General Public License v2
# This file is part of Checkmk (https://checkmk.com). It is subject to the terms and
# conditions defined in the file COPYING, which is part of this source code package.

from __future__ import annotations

import enum
from typing import NamedTuple

from cmk.utils.type_defs import HostAddress, HostName

from cmk.fetchers import FetcherType


class HostKey(NamedTuple):
    hostname: HostName
    source_type: SourceType


class SourceType(enum.Enum):
    """Classification of management sources vs regular hosts"""

    HOST = enum.auto()
    MANAGEMENT = enum.auto()


class SourceInfo(NamedTuple):
    hostname: HostName
    ipaddress: HostAddress | None
    ident: str
    fetcher_type: FetcherType
    source_type: SourceType

File: packages/cmk-frontend-vue/src/graph-designer/components/type_defs.ts
/**
 * Copyright (C) 2024 Checkmk GmbH - License: Checkmk Enterprise License
 * This file is part of Checkmk (https://checkmk.com). It is subject to the terms and
 * conditions defined in the file COPYING, which is part of this source code package.
 */

interface Element {
  ident: string
  title: string
}

export interface Topic {
  ident: string
  title: string
  elements: Element[]
}

export interface SpecLineType {
  line: string
  area: string
  stack: string
}
